src/main/java/auxiliary/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package auxiliary;
+
+import core.basesyntax.Circle;
+import core.basesyntax.IsoscelesTrapezoid;
+import core.basesyntax.Rectangle;
+import core.basesyntax.RightTriangle;
+import core.basesyntax.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int number = random.nextInt(TypeOfFigure.values().length);
+        if (number == 1) {
+            return new Square(getRandNumber());
```

Comment:
we should pass number and color to the contractor
```suggestion
            return new Square(getRandomNumber(), colorSupplier.getRandomColor());
```